backend/src/hmc-api/fetcher/utils.ts

```diff
@@ -0,0 +1,88 @@
+import * as APIv4 from "hyperschedule-shared/api/v4";
+import { DATA_DIR_PATH } from "hyperschedule-data";
+import type { Endpoint, Endpoints } from "./types";
+import { Params } from "./types";
+import { resolve, dirname } from "path";
+import { readFile, writeFile, mkdir } from "fs/promises";
+import { CourseFiles } from "../course";
+import { endpoints } from "./endpoints";
+import { createLogger } from "../../logger";
+
+const logger = createLogger("hmc.fetcher.utils");
+
+/**
+ * since each endpoint requires different parameter, we use this to generate all of them
+ */
+export function computeParams(term: APIv4.TermIdentifier): Params {
+    return Params.parse({
+        session: APIv4.stringifyTermIdentifier(term),
+        year: term.year.toString(),
+        catalog: `UG${
+            (term.term === APIv4.Term.fall ? term.year : term.year - 1) % 100
+        }`,
+    } satisfies Params);
+}
+
+export function computeFilePath(
+    endpoint: Endpoint,
+    term: APIv4.TermIdentifier,
+) {
+    return resolve(
+        DATA_DIR_PATH,
+        APIv4.stringifyTermIdentifier(term),
+        endpoint.saveAs,
+    );
+}
+
+/**
+ * loads a static file from the data directory corresponding to the endpoint
+ */
+export async function loadStatic(
+    endpoint: Endpoint,
+    term: APIv4.TermIdentifier,
+) {
+    return readFile(computeFilePath(endpoint, term));
+}
+
+/**
+ * saves the result of an endpoint call to the data directory
+ */
+export async function saveStatic(
+    content: string,
+    endpoint: Endpoint,
+    term: APIv4.TermIdentifier,
+) {
+    const path = computeFilePath(endpoint, term);
+
+    // TODO: figure out somewhere else to call this so we don't try to run mkdir for every new file
+    await mkdir(dirname(path), { recursive: true, mode: 0o700 });
+
+    const buf = Buffer.from(content, "utf-8");
+    // writes the content of the file using utf-8 encoding, even if the underlying data might not be
+    // properly encoded. that's why we have the fixEncoding function
+    await writeFile(path, buf, {
+        encoding: "utf-8",
+    });
+    logger.info(`File ${path} written (${(buf.length / 1024).toFixed(2)} KB)`);
+}
+
+export async function loadCourseFiles(
```

Comment:
once we have a real database up and running this code becomes irrelevant, right?

Reply:
No, because we still need to use the course files for unit testing and initializing a local dev database